pkg/resources/object_renaming_acceptance_test.go

```diff
@@ -0,0 +1,1366 @@
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: resource.ComposeAggregateTestCheckFunc(),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModels(t, databaseConfigModel, schemaModelConfig),
+			},
+			{
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("snowflake_database.test", plancheck.ResourceActionUpdate),
+						plancheck.ExpectResourceAction("snowflake_schema.test", plancheck.ResourceActionNoop),
+					},
+				},
+				Config:      config.FromModels(t, databaseConfigModelWithNewId, schemaModelConfig),
+				ExpectError: regexp.MustCompile("does not exist or not authorized"),
+			},
+		},
+	})
+}
+
+func TestAcc_ShallowHierarchy_IsInConfig_RenamedInternally_WithoutDependency_AfterRenameSchemaReferencingNewDatabaseName(t *testing.T) {
+	// Error happens during schema's Read operation and then Delete operation (schema cannot be removed).
+	t.Skip("Not able to handle the error produced by Delete operation that results in test always failing")
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableObjectRenamingTest)
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+
+	databaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	newDatabaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	schemaName := acc.TestClient().Ids.Alpha()
+
+	databaseConfigModel := model.Database("test", databaseId.Name())
+	databaseConfigModelWithNewId := model.Database("test", newDatabaseId.Name())
+
+	schemaModelConfig := model.Schema("test", databaseId.Name(), schemaName)
+	schemaModelConfigWithNewDatabaseId := model.Schema("test", newDatabaseId.Name(), schemaName)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.Schema),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModels(t, databaseConfigModel, schemaModelConfig),
+			},
+			{
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("snowflake_database.test", plancheck.ResourceActionUpdate),
+						plancheck.ExpectResourceAction("snowflake_schema.test", plancheck.ResourceActionDestroyBeforeCreate),
+					},
+				},
+				Config:      config.FromModels(t, databaseConfigModelWithNewId, schemaModelConfigWithNewDatabaseId),
+				ExpectError: regexp.MustCompile("does not exist or not authorized"),
+			},
+		},
+	})
+}
+
+func TestAcc_ShallowHierarchy_IsInConfig_RenamedInternally_WithDependsOn_AfterRenameSchemaReferencingOldDatabaseName(t *testing.T) {
+	// Error happens during schema's Read operation and then Delete operation (schema cannot be removed).
```

Comment:
how about having the destroy step only? would it help at all? (probably not, but worth asking)

Reply:
Yeah, I tried, but this step will also fail because it's the same as removing the resource after the test. Or maybe there's some other way, but I had too many tests to write to spend more time on this.